src/libdrakvuf/vmi.c

```diff
@@ -1162,18 +1214,24 @@ bool inject_trap_pa(drakvuf_t drakvuf,
         }
     }
 
-    if ( !inject_trap_mem(drakvuf, &container->breakpoint.guard, 0) )
+    if ( !inject_trap_mem(drakvuf, &container->breakpoint.guard, 0, drakvuf->altp2m_idx) )
     {
         PRINT_DEBUG("Failed to create guard trap for the breakpoint!\n");
         goto err_exit;
     }
 
-    if ( !inject_trap_mem(drakvuf, &container->breakpoint.guard2, 1) )
+    if ( !inject_trap_mem(drakvuf, &container->breakpoint.guard2, 1, drakvuf->altp2m_idx) )
     {
         PRINT_DEBUG("Failed to create guard2 trap for the breakpoint!\n");
         goto err_exit;
     }
 
+    if ( !inject_trap_mem(drakvuf, &container->breakpoint.guard3, 0, drakvuf->altp2m_idrx) )
```

Comment:
I would assume this is only needed if the context-based feature is used.

Reply:
It could be used in case we allow the plugins to toggle the context-based feature, in that case the breakpoints has to be set in order to make the toggling process easier.